src/MSBuildBinLogQuery/Component/BuildComponent.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.Build.Logging.Query.Messaging;
+
+namespace Microsoft.Build.Logging.Query.Component
+{
+    public abstract class BuildComponent
+    {
+        public IList<Message> Messages => _messages.ToImmutableList();
+        public IList<Warning> Warnings => _warnings.ToImmutableList();
+        public IList<Error> Errors => _errors.ToImmutableList();
+        public IList<Message> AllMessages => _allMessages.ToImmutableList();
+        public IList<Warning> AllWarnings => _allWarnings.ToImmutableList();
+        public IList<Error> AllErrors => _allErrors.ToImmutableList();
```

Comment:
What is the distinction between `Errors` and `AllErrors`? It looks like it's that `AllErrors` is recursive? Please doc-comment this.